vortex-array/src/array/varbin/mod.rs

```diff
@@ -44,9 +44,22 @@ impl VarBinArray {
         dtype: DType,
         validity: Validity,
     ) -> VortexResult<Self> {
-        if !offsets.dtype().is_int() || offsets.dtype().is_nullable() {
-            vortex_bail!(MismatchedTypes: "non nullable int", offsets.dtype());
-        }
+        let offsets_ptype = match offsets.dtype() {
+            DType::Primitive(
```

Comment:
nit: kind of the same statement as before, IMO this match statement is fairly complicated => i.e., less readable than the imperative check followed by `let offsets_ptype = PType::try_from(offsets.dtype()).vortex_unwrap()`, but I don't feel super strongly about it